interactive_shell_handler.c

```diff
@@ -0,0 +1,150 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <windows.h>
+#include <psapi.h>
+#include <tchar.h>
+#include "interactive_shell_handler.h"
+#include "utils.h"
+
+// forward declarations
+void perform_command(char *command);
+char** split_command(char *command);
+
+void handle_copy(char* src, char* dst) {
+    int exitCode = 0;
+
+    exitCode = copy_file(src, dst);
+    if (exitCode == 0) {
+        printf("copied source file to destination");
+    } else {
+        printf("failed to copy source file to destination");
+    }
+}
+
+void handle_move(char* src, char* dst) {
+    int exitCode = copy_file(src, dst);
+    if (exitCode == 0) {
+        // delete src file 
+        exitCode = remove(src);
+        if (exitCode == 0) {
+            printf("moved file successfully");
+        } else {
+            printf("failed to moved file");
+        }
+    }
+}
+
+void print_process_name_and_id (DWORD processId) {
+    TCHAR processName[MAX_PATH] = "<unknown>";
+    HANDLE hProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, processId);
+    if (hProcess != NULL) {
+        HMODULE hMod;
+        DWORD cbNeeded;
+
+        if(EnumProcessModulesEx(hProcess, &hMod, sizeof(hMod), &cbNeeded, LIST_MODULES_DEFAULT)) {
+            GetModuleBaseName(hProcess, hMod, processName, sizeof(processName)/sizeof(TCHAR));
+        }
+    }
+
+    printf("%s\t(PID: %u)\n", processName, processId);
+
+    CloseHandle(hProcess);
+}
+
+void handle_tasklist() {
+    #define getAProcessesSize(mult) (sizeof(DWORD)*PROCESS_LIST_STARTING_LEN*mult)
+
+    DWORD* aProcesses;
+    DWORD cbNeeded, cProcesses;
+    int mult = 1;
+    int aProcesessesSize = getAProcessesSize(mult);
+
+    aProcesses = (DWORD*)malloc(aProcesessesSize);
```

Comment:
You're allocating a resource here.
Make sure all code paths lead to a an appropriate free call.